dietpi/func/dietpi-globals

```diff
@@ -646,79 +642,48 @@ $(ps f -eo pid,user,tty,cmd | grep -i [d]ietpi)"; then
 		# - This can then be used to increase/decrease size of WHIP_SIZE_Z automatically.
 		else
 
-			if [[ $WHIP_MESSAGE ]]; then
-
-				local character_count_per_line=$(echo -e "$WHIP_MESSAGE" | awk '{ print length }')
-
-				# - Calculate lines required
-				while read -r line
-				do
-
-					((lines_required_whip_y++))
-
-					# - Calculate for addition required lines, if the text is longer than X
-					while (( $line > ( $WHIP_SIZE_X - $whip_x_internal_margin ) ));	do
-
-						line=$(( $line - ( $WHIP_SIZE_X - $whip_x_internal_margin ) ))
-						((lines_required_whip_y++))
-
-					done
+			# - Calculate lines required, including addition lines, if the text is longer than X
+			[[ $WHIP_MESSAGE ]] && echo -e "$WHIP_MESSAGE" | while read -r line
+			do
 
-				done <<< "$character_count_per_line"
+				(( lines_required_whip_y += 1 + ${#line} / ( $WHIP_SIZE_X - $whip_x_internal_margin) ))
 
-			fi
+			done
 
 			#	Process final whiptail size
 			#	Add internal whiptail margins to end line total
-			lines_required_whip_y=$(( $lines_required_whip_y + $whip_y_internal_margin ))
+			(( lines_required_whip_y += $whip_y_internal_margin ))
 
 		fi
 
 		#Calculate Z
 		# - G_WHIP_MENU_ARRAY
 		if (( $input_mode == 1 )); then
 
-			lines_required_whip_z=$(( ${#G_WHIP_MENU_ARRAY[@]} / 2 ))
-			# - Prevent single entries from being divided below 1, eg: 1 / 2 = 0 in bash
-			if (( $lines_required_whip_z < 0 )); then
-
-				lines_required_whip_z=1
-
-			fi
+			# - ( ${#array} + 1 ) to round up on uneven array entries
+			lines_required_whip_z=$(( ( ${#G_WHIP_MENU_ARRAY[@]} + 1 ) / 2 ))
 
 			# - Requires additional line on Y
 			((lines_required_whip_y++))
 
 			#Auto length for ─
 			#	Get max length of all the lines in odd number array 1st | '' 'this one'
 			local i=0
-			local index=0
 			local character_count_max=0
-			for (( i=0; i<${#G_WHIP_MENU_ARRAY[@]}; i++ ))
+			for (( i=0; i<${#G_WHIP_MENU_ARRAY[@]}; i+=2 ))
```

Comment:
@MichaIng 

Nice 👍

Reply:
@Fourdee 
Your comment as well. Lol found the `var+=$value` syntax already but did not use it there. Makes arithmetic assignments again simpler 😃.